openedx/features/discounts/tests/test_applicability.py

```diff
@@ -18,8 +24,26 @@ def setUp(self):
         super(TestApplicability, self).setUp()
         self.user = UserFactory.create()
         self.course = CourseFactory.create(run='test', display_name='test')
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='verified')
 
     def test_can_receive_discount(self):
         # Right now, no one should be able to receive the discount
-        applicability = can_receive_discount(user=self.user, course_key_string=self.course.id)
+        applicability = can_receive_discount(user=self.user, course=self.course)
+        self.assertEqual(applicability, False)
+
+    @override_waffle_flag(FIRST_PURCHASE_OFFER_BANNER_DISPLAY, active=True)
```

Comment:
You should not need to set this flag here -- whether it is displayed does not matter here.